src/Kestrel.Core/Internal/Infrastructure/HttpCharacters.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure
+{
+    public static class HttpCharacters
+    {
+        public static readonly int TableSize = 128;
+        public static readonly bool[] AlphaNumeric;
```

Comment:
Exposing public arrays lets consumers mutate their contents. 
Could they be exposed as `ReadOnlySpan<bool>` instead?h

Reply:
Actually, these probably shouldn't be exposed as public. In fact, these definitions probably belong to HttpAbstractions. However, since it's only being used in Kestrel for now, I'll set the accessibility to internal.